### ISSUE TITLE
allow toc link markup to be preserved?

### ISSUE THREAD

**Issue by mattgarrish** [w3c/wpub#414]
Closing those old toc issues jarred my memory that I don't think we ever took up the question about whether a text label has to be generated from the toc links or whether the child html content can be preserved.

Right now the algorithm references the accessible name calculation algorithm for retrieving a text label from the markup, but we may want to add the option to use the child markup itself?

Or it might be best to end the algorithm with the extraction of the child markup and leave it to the user agent to determine how best to present the label, with perhaps only guidance to use the accname algorithm when only a simple text label is desired?

**Comment by iherman** [comment]
One of the reason we opted for the HTML based TOC is i18n. Ie, the chapter title may include ruby markup. I think that should be preserved, for example.

(I am not sure how this relates to the accessible name calculation, to be honest.)

**Comment by mattgarrish** [comment]
Right, I think we went with text strings so that if the toc was also embedded directly in the manifest then the resulting structure would be the same regardless of how it was authored. Given embedding is not a consideration anymore, I think we should reconsider requiring text strings as the outcome.

**Comment by dauwhe** [comment]
The ability of HTML to express more than text strings is important. Quite a bit of inline markup is semantically meaningful. Let's not give it up!

**Comment by HadrienGardeur** [comment]
While we might "allow" it, I doubt that UAs will retain anything more than plain text. 

This has been discussed ad nauseam, but as @dauwhe has experimented himself, most UAs simply sanitize every string in a TOC.

**Comment by mteixeira-wwn** [comment]
Here are a couple of examples of real EPUB 3.0.1 navigation markup. I mentioned in our call that we accept that these are nonstandard practices today, but I hope they're helpful insofar as they provide examples of eccentric, yet reasonably common, use cases in educational publishing.

> The examples below use certain attributes (`epub:type`, `data-`) that are used by our internal publishing automation tools or for cross-product interop. They can be ignored.

- A biochemistry book where units are organizational taxonomic labels. Chapter content documents still needed to be grouped by units. This is an example of `nav` entries without associated content document links.

```html
<li><span>INTRODUCTION</span>
    <ol>
        <li><a href="content/chapter01.xhtml" data-spine-index="10" data-nav-index="1" epub:type="chapter" data-page-start="2" data-page-end="5">Chapter 1: The Nature of Science</a>
            <ol>
                <li><a href="content/chapter01-01.xhtml" data-spine-index="11" data-nav-index="1.1" data-page-start="6" data-page-end="7">Bat Crazy</a></li>
                <li><a href="content/chapter01-02.xhtml" data-spine-index="12" data-nav-index="1.2" data-page-start="8" data-page-end="9">Prove Me Wrong</a></li>
                <li><a href="content/chapter01-03.xhtml" data-spine-index="13" data-nav-index="1.3" data-page-start="10" data-page-end="13">Catching the Culprit</a></li>
                <li><a href="content/chapter01-04.xhtml" data-spine-index="14" data-nav-index="1.4" data-page-start="14" data-page-end="16">No End in Sight</a></li>
                <li><a href="content/chapter01-review.xhtml" data-spine-index="15" data-nav-index="1.5" epub:type="practice" data-page-start="17" data-page-end="19">Review</a></li>
            </ol>
        </li>
        ( . . . )
    </ol>
</li>
<li><span>UNIT 1: CELLS</span>
    <ol>
        <li><a href="content/chapter03.xhtml" data-spine-index="24" data-nav-index="3" epub:type="chapter" data-page-start="38" data-page-end="43">Chapter 3: Chemistry of Life</a>
            <ol>
                <li><a href="content/chapter03-01.xhtml" data-spine-index="25" data-nav-index="3.1" data-page-start="44" data-page-end="44">One Picture, a Thousand Experiments</a></li>
                <li><a href="content/chapter03-02.xhtml" data-spine-index="26" data-nav-index="3.2" data-page-start="45" data-page-end="47">The World of Water</a></li>
                <li><a href="content/chapter03-03.xhtml" data-spine-index="27" data-nav-index="3.3" data-page-start="48" data-page-end="48">The Smell of Success</a></li>
                <li><a href="content/chapter03-04.xhtml" data-spine-index="28" data-nav-index="3.4" data-page-start="49" data-page-end="51">Getting the Right Mix</a></li>
                <li><a href="content/chapter03-05.xhtml" data-spine-index="29" data-nav-index="3.5" data-page-start="52" data-page-end="55">Life’s First Steps</a></li>
                <li><a href="content/chapter03-06.xhtml" data-spine-index="30" data-nav-index="3.6">Fifty More Years</a></li>
                <li><a href="content/chapter03-review.xhtml" data-spine-index="31" data-nav-index="3.7" epub:type="practice" data-page-start="56" data-page-end="57">Review</a></li>
            </ol>
        </li>
        ( . . . )
    </ol>
</li>
```

- A music theory book in which the table of contents includes inline music notation. In the `nav`, we opted for inline SVGs where regular HTML markup (e.g., `sup` and `sub` elements) wasn't viable. This content is only delivered through our platform, so we were not concerned about presentation in other readers. The Norton Ebook Reader is a Web application and renders navigation documents as HTML, not plaintext. If we were to provide this content externally, we'd need to come up with editorially equivalent ways to describe the notation in plaintext, which in many cases would involve complete rewrites and result in disparate experiences in cross-media settings (e.g., classrooms). The two samples below are a small representation of simple markup—the SVGs can be rather complex.

```html
<li><a href="xhtml/Chapter10.html#ch10">10 The Dominant Seventh Chord: V<img class="nav__notation--fb7" src="images/fb7.svg" alt="figured bass, seventh chord" /></a>
    <ol>
        <li><a href="xhtml/Chapter10.html#ch10-0">V<img class="nav__notation--fb7" src="images/fb7.svg" alt="figured bass, seventh chord" /> and Tendency Tones</a></li>
        <li><a href="xhtml/Chapter10-1.html#ch10-1">Omitting the Fifth of V<img class="nav__notation--fb7" src="images/fb7.svg" alt="figured bass, seventh chord" /> or I in Four-Part Harmony</a></li>
        <li><a href="xhtml/Chapter10-2.html#ch10-2">V to V<img class="nav__notation--fb7" src="images/fb7.svg" alt="figured bass, seventh chord" /> (V<sup>8&#8211;7</sup>)</a></li>
        <li><a href="xhtml/Chapter10-3.html#ch10-3">V<img class="nav__notation--fb7" src="images/fb7.svg" alt="figured bass, seventh chord" /> at Cadences</a></li>
        <li><a href="xhtml/Chapter10-4.html#ch10-4">Melodic Patterns</a></li>
        <li><a href="xhtml/Chapter10-5.html#ch10-5">Points for Review</a></li>
        <li><a href="xhtml/Chapter10-5.html#ch10-6">Test Yourself</a></li>
    </ol>
</li>
<li><a href="xhtml/Chapter11.html#ch11">11 I<sup>6</sup> and V<sup>6</sup></a>
    <ol>
        <li><a href="xhtml/Chapter11.html#ch11-0">I<sup>6</sup></a></li>
        <li><a href="xhtml/Chapter11-1.html#ch11-1">V<sup>6</sup></a></li>
        <li><a href="xhtml/Chapter11-2.html#ch11-2">Points for Review</a></li>
        <li><a href="xhtml/Chapter11-2.html#ch11-3">Test Yourself</a></li>
    </ol>
</li>
```

**Comment by mattgarrish** [comment]
Closing this as it was addressed in #452.
